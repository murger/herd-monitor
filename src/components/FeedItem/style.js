import styled from 'styled-components';

export const Item = styled.li`
  display: flex;
  flex-wrap: wrap;
	padding: 8px;
  border-radius: 4px;
  cursor: pointer;

  &:hover {
    background-color: ${props => props.theme.offwhite};
  }

  span {
    padding: 2px 0;
    padding-right: 8px;
    box-sizing: border-box;
  }
`;

export const Id = styled.span`
  flex-basis: 72px;
  color: ${props => props.theme.offblack};
  border-radius: 4px;

  @media (max-width: 768px) {
    flex-basis: 25%;
  }
`;

export const Date = styled.span`
  flex-basis: 98px;
  color: ${props => props.theme.darkgray};

  @media (max-width: 768px) {
    flex-basis: 75%;
  }
`;

export const Price = styled.span`
  flex-grow: 1;
  color: ${props => props.theme.gray};

  del {
    padding-left: 8px;
    color: ${props => props.theme.lightgray};
  }
`;

export const Status = styled.span`
  && {
    padding-left: 8px;
    padding-right: 8px;
  }

  color: ${props => props.theme.darkgray};
  background-color: ${props => props.theme.offwhite};
  border-radius: 4px;

  ${Item}:hover & {
    color: white;
    background-color: ${props => props.theme.getStatusColour(props.type)};
  }
`;
