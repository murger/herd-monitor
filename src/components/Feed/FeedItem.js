import React from 'react';
import { format, differenceInCalendarDays } from 'date-fns';
import { Item, Id, Date, Price, Status } from './style';
import translateStatus from '../../utils/translateStatus';
import formatCurrency from '../../utils/formatCurrency';

const FeedItem = ({ item, ...props }) => (
  <Item {...props}>
    <Id>#{item.id}</Id>
    <Date>
      <span>{format(item.fromDate, "D MMM 'YY")}</span>
      <i>({differenceInCalendarDays(item.toDate, item.fromDate)}d)</i>
    </Date>
    <Price>
      {formatCurrency(item.price, item.currency)}
      {item.totalDiscount > 0 &&
        <del>{formatCurrency(item.totalDiscount, item.currency)}</del>
      }
    </Price>
    <Status type={item.status}>
      {translateStatus(item.status)}
    </Status>
  </Item>
);

export default FeedItem;
