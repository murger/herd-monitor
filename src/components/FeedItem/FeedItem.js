import React from 'react';
import { Item } from './style';

const FeedItem = ({ transaction }) => (
	<Item>
		#{transaction.lenderId} &rarr; #{transaction.borrowerId}
	</Item>
);

export default FeedItem;
