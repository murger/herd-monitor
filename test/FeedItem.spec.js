import React from 'react';
import renderer from 'react-test-renderer';
import FeedItem from '../src/components/FeedItem';

it('renders correctly', () => {
  const data = {
    'id': 35886,
    'lenderId': 8676,
    'borrowerId': 68115,
    'itemId': 3949,
    'fromDate': '2018-02-15 00:00:00+00',
    'toDate': '2018-02-16 00:00:00+00',
    'status': 'FL_APPROVED',
    'promocode': null,
    'creditUsed': 0,
    'price': 2025,
    'totalDiscount': 0,
    'currency': 'GBP',
  };
  const tree = renderer.create(<FeedItem item={data} />).toJSON();

  expect(tree).toMatchSnapshot();
});
