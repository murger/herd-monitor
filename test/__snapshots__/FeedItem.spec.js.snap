// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the correctly 1`] = `
<li
  className="sc-bdVaJa cISSwi"
>
  <span
    className="sc-bwzfXH zkVWP"
  >
    #
    35886
  </span>
  <span
    className="sc-htpNat eAfzhL"
  >
    <span>
      15 Feb '18
    </span>
    <i>
      (
      1
      d)
    </i>
  </span>
  <span
    className="sc-bxivhb dzoBoK"
  >
    Â£2,025.00
  </span>
  <span
    className="sc-ifAKCX cTbvsb"
    type="FL_APPROVED"
  >
    Approved
  </span>
</li>
`;
